yaks/lib/yaks/reader/json_api.rb

```diff
@@ -0,0 +1,49 @@
+module Yaks
+  module Reader
+    class JsonAPI
+      def call(parsed_json, env = {})
+        attributes = parsed_json['data'].first.dup
+        links = attributes.delete('links') || {}
+        linked = parsed_json['linked'].nil? ? {} : parsed_json['linked'].dup
+        embedded   = convert_embedded(links, linked)
+        Resource.new(
+            type: Util.singularize(attributes.delete('type')[/\w+$/]),
+            attributes: Util.symbolize_keys(attributes),
+            subresources: embedded,
+            links: []
+        )
+      end
+
+      def convert_embedded(links, linked)
+        links.flat_map do |rel, link_data|
+          # If this is a compound link, the link_data will contain either
+          # * 'type' and 'id' for a one to one
+          # * 'type' and 'ids' for a homogeneous to-many relationship
+          # * 'data' being an array where each member has 'type' and 'id' for heterogeneous
```

Comment:
Thank you for this! This complexity at the reading side is one of the things that IMO make JSON-API much too complex. Cool that you've actually gone to the trouble of catering for all valid cases. :)